src/store/index.js

```diff
@@ -9,17 +9,62 @@ import type { Data, Message } from './types';
 
 const store: Data = require('./emails.json');
 
-/**
- * You might find it useful to implement some functions that make accessing
- * deeply nested attributes of Messages or Threads...
- */
+function getNewMessageIds() {
+  return store.mailboxes.INBOX.threadIds;
+}
+
+function getNewMessages(newMessageIds) {
+  let newMessages = [];
+  for (let id = 0; id < newMessageIds.length; id += 1) {
+    const rawMessageObj = store.messages[newMessageIds[id]];
+    const refinedMessageObj = {
+      Snippet: rawMessageObj.snippet,
+      Date: '',
+      From: '',
+      To: '',
+      Subject: '',
+    };
+    for (let header of rawMessageObj.payload.headers) {
+      refinedMessageObj[header.name] = header.value;
+    };
+    newMessages.push(refinedMessageObj);
+  }
```

Comment:
This is a very creative solution. In a CS sense, you’re building an index of the headers array. I like what you’re going for but I have a few tips that might make your intentions more clear.

I understand what you're going for with the variable named `refinedMessageObj` but by the end of this function that object has a key for every header—of which there could be many that you don’t care about. If you want to build an index by looping through the headers, I would recommend doing that in one step and then choosing the headers you care about in a separate step.

Additionally, it seems that your purpose for setting out `Date: '', From: '', ...` is to provide defaults, but this would be more clear if you, again, broke that out into a clear “step” of your algo. Maybe something like `const withDefaults = { ...defaultHeaders, ...headerIndex }`

I think those things would help make your intent more clear here. But I wonder, did you consider other strategies as well? Building an index is a strategy that is well suited for situations where you've got a very large array, a chance of accessing any item, and a plan to read from it many many times. But here you're only going to be accessing a few known properties and you'll only be reading from the index four times per email. Is there something you could do to reduce the amount of unnecessary work you're doing by indexing all the headers you won't ever need?